File: CustomFeedCpp/WidgetFeedProvider.cpp
#include "pch.h"
#include "WidgetFeedProvider.h"

WidgetFeedProvider::WidgetFeedProvider()
{
}

void WidgetFeedProvider::OnCustomQueryParametersRequested(CustomQueryParametersRequestedArgs args)
{
	printf("Custom query parmaeters requested for feed provider %ls!\n", args.FeedProviderDefinitionId().c_str());
	FeedManager::GetDefault().SetCustomQueryParameters(CustomQueryParametersUpdateOptions(args.FeedProviderDefinitionId(), L"?widgets=True"));
}

void WidgetFeedProvider::OnFeedDisabled(FeedDisabledArgs args)
{
	printf("Feed %ls disabled!\n", args.FeedDefinitionId().c_str());
}

void WidgetFeedProvider::OnFeedEnabled(FeedEnabledArgs args)
{
	printf("Feed %ls enabled!\n", args.FeedDefinitionId().c_str());
}

void WidgetFeedProvider::OnFeedProviderDisabled(FeedProviderDisabledArgs args)
{
	printf("Feed provider %ls disabled!\n", args.FeedProviderDefinitionId().c_str());
}

void WidgetFeedProvider::OnFeedProviderEnabled(FeedProviderEnabledArgs args)
{
	printf("Feed provider %ls enabled!\n", args.FeedProviderDefinitionId().c_str());
}

File: CustomFeedProvider/WidgetFeedProvider.cs
using Microsoft.Windows.Widgets.Feeds.Providers;
using System.Runtime.InteropServices;
using System.Runtime.InteropServices.Marshalling;

namespace CustomFeedProvider;

[Guid("606dbaa8-4054-4865-8ddf-4d518a770a4c")]
[ComVisible(true)]
[ComDefaultInterface(typeof(IFeedProvider))]
public partial class WidgetFeedProvider : IFeedProvider
{
    public void OnCustomQueryParametersRequested(CustomQueryParametersRequestedArgs args)
    {
        Console.WriteLine("Custom query parameters requested for feed provider: {0}", args.FeedProviderDefinitionId);
    }

    public void OnFeedDisabled(FeedDisabledArgs args)
    {
        Console.WriteLine("Feed disabled: {0}, provider: {1}", args.FeedDefinitionId, args.FeedProviderDefinitionId);
    }

    public void OnFeedEnabled(FeedEnabledArgs args)
    {
        Console.WriteLine("Feed enabled: {0}, provider: {1}", args.FeedDefinitionId, args.FeedProviderDefinitionId);
    }

    public void OnFeedProviderDisabled(FeedProviderDisabledArgs args)
    {
        Console.WriteLine("Feed provider disabled: {0}", args.FeedProviderDefinitionId);
    }

    public void OnFeedProviderEnabled(FeedProviderEnabledArgs args)
    {
        Console.WriteLine("Feed provider enabled: {0}", args.FeedProviderDefinitionId);
    }
}